src/festim/hydrogen_transport_problem.py

```diff
@@ -1209,18 +1209,32 @@ def mixed_term(u, v, n):
         self.forms = dolfinx.fem.form(
             [subdomain.F for subdomain in self.volume_subdomains],
             entity_maps=entity_maps,
+            jit_options={
+                "cffi_extra_compile_args": ["-O3", "-march=native"],
+                "cffi_libraries": ["m"],
+            },
+        )
+        self.J = dolfinx.fem.form(
+            J,
+            entity_maps=entity_maps,
+            jit_options={
+                "cffi_extra_compile_args": ["-O3", "-march=native"],
+                "cffi_libraries": ["m"],
+            },
         )
-        self.J = dolfinx.fem.form(J, entity_maps=entity_maps)
 
     def create_solver(self):
-        self.solver = NewtonSolver(
+        self.solver = BlockedNewtonSolver(
             self.forms,
-            self.J,
             [subdomain.u for subdomain in self.volume_subdomains],
+            J=self.J,
             bcs=self.bc_forms,
-            max_iterations=self.settings.max_iterations,
             petsc_options=self.petsc_options,
         )
+        self.solver.max_iterations = (self.settings.max_iterations,)
```

Comment:
Why does this need to be a tuple?

Reply:
It shouldn’t, no clue how that snuck in there